homeassistant/components/reolink/update.py

```diff
@@ -68,6 +68,25 @@ def latest_version(self) -> str | None:
 
         return self.coordinator.data.version_string
 
+    @property
+    def supported_features(self) -> UpdateEntityFeature:
+        """Flag supported features."""
+        supported_features = UpdateEntityFeature.INSTALL
+        if isinstance(self.coordinator.data, NewSoftwareVersion):
+            supported_features |= UpdateEntityFeature.RELEASE_NOTES
```

Comment:
Why not set that this always support release notes instead of recalculating based if there are a new version available or not?

Reply:
Because if a user blocks internet acces for the Reolink device (commonly done and supported by this integration), the online check on the reolink.com server will fail and there will not be any Release notes available.
In that case the self.coordinator.data will be False (no update) or a string (version) instead of the NewSoftwareVersion class.

So we have to check dynamically.